### PR TITLE
Feature: Added an icon to the system tray

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

Users doesn't have a proper way to kill the app when running on the background. Added system tray icon to handle killing app by the app itself.

## Task Checklist

- [x] Added utility classes for the Taskbar
- [x] Added functionalities to kill/restart
- [ ] ~~Added functionalities to resume~~

## Steps To Validate Changes

1. Launch the app.
2. Enable background process running.
3. Close window.
4. See notification area on the taskbar.

## Known Issues

_N/A_

## PR Checklist

- [x] **Development**: #14240
- [x] **Approval**: Have discussed with and got approval from the team[^1]
- [x] **Tests**: Tested accessibility on the local end
- [x] **Deployment**: Deployed the app on the local end
- [x] **Localization**: Modified en-US string resources[^2]
- [ ] **Co-authors**: _N/A_

## Screenshots

image

[^1]: If the request hasn't been agreed by the team, this work might be rejected. Make sure that you get approval from the team before opening any PR related the request.
[^2]: If you removed any en-US string resources, make sure that there are no references of those resources. When you add a new en-US string resources, do not make any changes on other languages' resources; [Crowdin](https://crowdin.com/project/files-app) will do that, instead.

### CODE REVIEW

**Comment by yaira2** [comment]
> Added functionalities to kill/resume

Resume isn't needed for now as that can already be done from launching the app normally. The main part is indicating when the process is open and providing a way to close/restart.

**Comment by 0x5bfa** [comment]
All implementation requirements then have met.

Restarting and exitting have been implemented already and works very well.

**Comment by yaira2** [comment]
Is this ready for review?

**Comment by 0x5bfa** [comment]
Yes.

**Comment by yaira2** [comment]
> Yes. You also can open docs page and submit a bug from the tray.

I like the idea of linking to the docs but let's take out the option to submit a bug report. It's best to keep that option to the about page which has the version number and log file.

File: src/Files.App/Strings/en-US/Resources.resw [hunk 1]
```diff
@@ -3662,4 +3662,10 @@
   <data name="FailedToRotateImage" xml:space="preserve">
     <value>Failed to rotate the image</value>
   </data>
+  <data name="Restart" xml:space="preserve">
+    <value>Restart Files</value>
```
Comment (yaira2) [inline src/Files.App/Strings/en-US/Resources.resw:3666]:
```suggestion
    <value>Restart</value>
```

File: src/Files.App/Strings/en-US/Resources.resw [hunk 2]
```diff
@@ -3662,4 +3662,10 @@
   <data name="FailedToRotateImage" xml:space="preserve">
     <value>Failed to rotate the image</value>
   </data>
+  <data name="Restart" xml:space="preserve">
+    <value>Restart</value>
+  </data>
+  <data name="Quit" xml:space="preserve">
+    <value>Quit Files</value>
```
Comment (yaira2) [inline src/Files.App/Strings/en-US/Resources.resw:3669]:
```suggestion
    <value>Quit</value>
```

File: src/Files.App/Utils/Taskbar/SystemTrayIcon.cs [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+
+		private void ShowContextMenu()
+		{
+			PInvoke.GetCursorPos(out var lpPoint);
+
+			DestroyMenuSafeHandle hMenu = PInvoke.CreatePopupMenu_SafeHandle();
+
+			// Generate the classic context menu
+			PInvoke.AppendMenu(hMenu, MENU_ITEM_FLAGS.MF_BYCOMMAND, WM_FILES_CONTEXTMENU_RESTART, "Documentation".GetLocalizedResource());
+			PInvoke.AppendMenu(hMenu, MENU_ITEM_FLAGS.MF_BYCOMMAND, WM_FILES_CONTEXTMENU_QUIT , "SubmitBugReport".GetLocalizedResource());
```
Comment (yaira2) [inline src/Files.App/Utils/Taskbar/SystemTrayIcon.cs:226]:
```suggestion
```

File: src/Files.App/Utils/Taskbar/SystemTrayIcon.cs [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+
+		private void OnRestartClicked()
+		{
+			Microsoft.Windows.AppLifecycle.AppInstance.Restart("");
+
+			Program.Pool.Release();
+			Environment.Exit(0);
+		}
+
+		private void OnQuitClicked()
```
Comment (yaira2) [inline src/Files.App/Utils/Taskbar/SystemTrayIcon.cs:267]: Will this save the open tabs or is this similar to killing the process from Task Manager?

Comment (0x5bfa) [reply src/Files.App/Utils/Taskbar/SystemTrayIcon.cs:266]: When closed the window, saving is done already.

> this similar to killing the process from Task Manager

Yeah. Just killing the process and threads.
Task Manager, I believe, uses taskkill command or WMI tho.

Comment (0x5bfa) [reply src/Files.App/Utils/Taskbar/SystemTrayIcon.cs:266]: > When closed the window, saving is done already.

That's a problem when resuming on the flip side.

Comment (yaira2) [reply src/Files.App/Utils/Taskbar/SystemTrayIcon.cs:266]: > When closed the window, saving is done already.

What if someone uses this option while the window is still open?

Comment (0x5bfa) [reply src/Files.App/Utils/Taskbar/SystemTrayIcon.cs:266]: Wdym?

Comment (yaira2) [reply src/Files.App/Utils/Taskbar/SystemTrayIcon.cs:266]: If the window is still open, do the tabs still get saved when using this option?

Comment (0x5bfa) [reply src/Files.App/Utils/Taskbar/SystemTrayIcon.cs:266]: Yes.

**Comment by 0x5bfa** [comment]
Removing the icon when resumed from the Start Menu was not working proeprly; however, fixed as of now.

**Review by yaira2 — APPROVED** [review]
LGTM

**Comment by yaira2** [comment]
@komprimiert you can turn off system tray icons from the Taskbar settings in Windows.

**Comment by komprimiert** [comment]
> @komprimiert you can turn off system tray icons from the Taskbar settings in Windows.

I can't with this program have you tried it yourself?

**Comment by yaira2** [comment]
It works for me
image

**Comment by komprimiert** [comment]
> It works for me image

but you know that I want to completely hide it? not just hiding it into the system tray instead of the taskbar

**Comment by komprimiert** [comment]
can you maybe add a feature to hide it completely?

**Comment by yaira2** [comment]
It's something we're discussing but haven't finalized yet

**Comment by komprimiert** [comment]
> It's something we're discussing but haven't finalized yet

is that on the task board?

**Pull request merged**